src/server/pfs/server/server_test.go

```diff
@@ -5015,8 +5015,9 @@ func TestPutObjectAsync(t *testing.T) {
 	w, err := client.PutObjectAsync([]*pfs.Tag{tag})
 	require.NoError(t, err)
 	expected := []byte(generateRandomString(30 * MB))
-	_, err = w.Write(expected)
+	n, err := w.Write(expected)
 	require.NoError(t, err)
+	require.Equal(t, len(expected), n)
```

Comment:
Is there a way we can test this that reproduces the actual corruption? This is basically just testing part of the required semantics of `Write`. I could see a class of bugs that still return the right length and a nil error but exhibit the corruption issue.

Reply:
I'm working on a test to confirm this. I agree that I should confirm that this fix addresses the actual user issue we are hitting. I'm not sure if we want this as a part of CI though because it requires a large number of output files.